File: mongodb/src/main/java/mongo_solutions/1f.js
db.Business.aggregate([
    {$lookup: {from: "Review", localField: "business_id", foreignField: "business_id", as: "Reviews"}},
    {$unwind: "$Reviews"},
    {$group: {_id: "$name", averageStars: {$avg: "$stars"}}},
    {$sort: {avg: -1}}
]);

File: mongodb/documentation/source_code/task6/1f.java
/*
private void executeTask1f() {
    AggregateIterable result = businessCollection.aggregate(
            asList(
                    lookup("Review", "business_id", "business_id", "Reviews"),
                    unwind("$Reviews"),
                    group("$name", avg("averageStars", "$stars")),
                    sort(descending("averageStars"))
            ));

    for (Object o : result) {
        System.out.println(o);
    }
}
*/